app/src/main/java/org/thoughtcrime/securesms/onboarding/LinkDeviceActivity.kt

```diff
@@ -1,230 +1,160 @@
-    val recoveryPhraseFragment = RecoveryPhraseFragment()
-
-    override fun getCount(): Int {
-        return 2
-    }
-
-    override fun getItem(index: Int): Fragment {
-        return when (index) {
-            0 -> recoveryPhraseFragment
-            1 -> {
-                val result = ScanQRCodeWrapperFragment()
-                result.delegate = activity
-                result.message = activity.getString(R.string.activity_link_device_qr_message)
-                result
-            }
-            else -> throw IllegalStateException()
+@Preview
+@Composable
+fun PreviewRecoveryPassword() = RecoveryPassword(state = LinkDeviceState())
+
+@Composable
+fun RecoveryPassword(state: LinkDeviceState, onChange: (String) -> Unit = {}, onContinue: () -> Unit = {}) {
+    Column(
+        modifier = Modifier.padding(horizontal = 60.dp)
+    ) {
+        Spacer(Modifier.weight(1f))
+        Row {
+            Text(stringResource(R.string.sessionRecoveryPassword), style = MaterialTheme.typography.h4)
+            Spacer(Modifier.width(6.dp))
+            Icon(
+                painter = painterResource(id = R.drawable.ic_shield_outline),
+                contentDescription = null,
+            )
         }
-    }
-
-    override fun getPageTitle(index: Int): CharSequence {
-        return when (index) {
-            0 -> activity.getString(R.string.activity_link_device_recovery_phrase)
-            1 -> activity.getString(R.string.activity_link_device_scan_qr_code)
-            else -> throw IllegalStateException()
+        Spacer(Modifier.size(28.dp))
+        Text(stringResource(R.string.activity_link_enter_your_recovery_password_to_load_your_account_if_you_haven_t_saved_it_you_can_find_it_in_your_app_settings))
+        Spacer(Modifier.size(24.dp))
+        SessionOutlinedTextField(
+            text = state.recoveryPhrase,
+            modifier = Modifier
+                .fillMaxWidth()
+                .contentDescription(R.string.AccessibilityId_recovery_phrase_input),
+            placeholder = stringResource(R.string.recoveryPasswordEnter),
+            onChange = onChange,
+            onContinue = onContinue,
+            error = state.error
+        )
+        Spacer(Modifier.size(12.dp))
+        state.error?.let {
+            Text(
+                it,
+                modifier = Modifier.contentDescription(R.string.AccessibilityId_error_message),
+                style = MaterialTheme.typography.baseBold,
+                color = MaterialTheme.colors.error
+            )
         }
+        Spacer(Modifier.weight(2f))
+        OutlineButton(
+            textId = R.string.continue_2,
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(horizontal = 64.dp, vertical = 20.dp)
+                .width(200.dp),
+            onClick = onContinue
```

Comment:
In the docs and in figma, the next screen is the "Message notifications" and THEN the loading animation. But in my test the animation came first followed by the "Message notifications" screen